tests/src/Kernel/Entity/SelectionHandlerTest.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/**
+ * @file
+ * Contains Drupal\Tests\og\Kernel\Entity\SelectionHandlerTest.
+ */
+
+namespace Drupal\Tests\og\Kernel\Entity;
+
+use Drupal\Component\Utility\Unicode;
+use Drupal\node\Entity\NodeType;
+use Drupal\KernelTests\KernelTestBase;
+use Drupal\og\Og;
+
+/**
+ * Tests entity reference selection plugins.
+ *
+ * @group og
+ */
+class SelectionHandlerTest extends KernelTestBase {
+
+  /**
+   * The selection handler.
+   *
+   * @var \Drupal\og\Plugin\EntityReferenceSelection\OgSelection.
+   */
+  protected $selectionHandler;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static $modules = ['system', 'user', 'field', 'entity_reference', 'node', 'og'];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    // Add membership and config schema.
+    $this->installConfig(['og']);
+    $this->installEntitySchema('og_membership');
+    $this->installEntitySchema('user');
+    $this->installEntitySchema('node');
+    $this->installSchema('system', 'sequences');
+
+    // Create a group.
+    NodeType::create([
+      'type' => $group_type = Unicode::strtolower($this->randomMachineName()),
+      'name' => $this->randomString(),
+    ])->save();
+
+    // Create a group content type.
+    NodeType::create([
+      'type' => $group_content_type = Unicode::strtolower($this->randomMachineName()),
```

Comment:
This is how it's done in core but there is an improvement i'd like to do: move part of the logic for setting up groups into traits so we could use this in other tests classes.